File: 2025-02-February-GFG-POTD/February 17 - k largest elements/Solution.py
import heapq

class Solution:
    def kLargest(self, arr, k):
        # Use a max heap by inverting values
        maxHeap = [-num for num in arr]
        heapq.heapify(maxHeap)

        result = []
        for _ in range(k):
            result.append(-heapq.heappop(maxHeap))
        return result


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        k = int(input())

        ob = Solution()
        ans = ob.kLargest(arr, k)

        print(*ans)
        print("~")

File: 2025-02-February-GFG-POTD/February 17 - k largest elements/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> kLargest(vector<int>& arr, int k) {
        // Use a max heap
        priority_queue<int> maxHeap(arr.begin(), arr.end());

        vector<int> result;
        for (int i = 0; i < k; i++) {
            result.push_back(maxHeap.top());
            maxHeap.pop();
        }
        return result;
    }
};

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (int i = 0; i < n; i++) cin >> arr[i];

        int k;
        cin >> k;

        Solution ob;
        vector<int> ans = ob.kLargest(arr, k);

        for (int num : ans) {
            cout << num << " ";
        }
        cout << "\n~\n";
    }
    return 0;
}
